src/Sentry/SentryOptions.cs

```diff
@@ -1358,14 +1358,38 @@ public void AddInAppExclude(string prefix)
     {
         if (InAppExclude == null)
         {
-            InAppExclude = new() { prefix };
+            InAppExclude = [prefix];
         }
         else
         {
             InAppExclude.Add(prefix);
         }
     }
 
+    /// <summary>
+    /// Add a regex pattern used to identify frames that are not 'InApp' in stacktraces.
+    /// </summary>
+    /// <param name="pattern">The regular expression to use to match stacktrace frames</param>
+    /// <remarks>
+    /// Sentry by default filters the stacktrace to display only application code.
+    /// A user can optionally click to see all which will include framework and libraries.
+    /// If a frame has not already been determined to be InApp per any of the prefixes or
+    /// patterns configured in <see cref="InAppInclude"/>, it will be considered InApp only
+    /// if it does not match any of the prefixes or patterns in <see cref="InAppExclude"/>.
+    /// </remarks>
+    public void AddInAppExcludeRegex(string pattern)
```

Comment:
Instead of parsing ourselves, I'd suggest adding `AddInAppExclude(Regex pattern)`

Reply:
That was originally how I had it... it means reminding people they should use the right options to get decent performance. When parsing ourselves, we ensure the RegEx is created with sensible options:
```
new Regex(pattern, RegexOptions.Compiled | RegexOptions.CultureInvariant)
```

If these Regex instances were going to be used in ways we couldn't anticipate, we'd  definitely be better off giving people more flexibility. Since we know that they will only be used to match namespaces (i.d. perdiod delimited [C# identifiers](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#643-identifiers)) I wondered if we could make the API simpler - so give people less rope to hang themselves but still be powerful enough to do what they need. 

I was a bit worried about the use of `RegexOptions.CultureInvariant`. But the [docs says](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-options#compare-using-the-invariant-culture) that only affects case insensitive matches (namespaces would be case sensitive).